pkg/config/config.go

```diff
@@ -249,10 +281,50 @@ func (c *MicroshiftConfig) ReadAndValidate(configFile string, flags *pflag.FlagS
 	// validate serviceCIDR
 	clusterDNS, err := getClusterDNS(c.Cluster.ServiceCIDR)
 	if err != nil {
-		klog.Fatalf("failed to get DNS IP: %v", err)
+		return fmt.Errorf("failed to get DNS IP: %v", err)
 	}
 	c.Cluster.DNS = clusterDNS
 
+	if len(c.SubjectAltNames) > 0 {
+		u, err := url.Parse(c.Cluster.URL)
+		if err != nil {
+			return fmt.Errorf("failed to parse cluster URL: %v", err)
+		}
+		if u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
+			if stringSliceContains(c.SubjectAltNames, "localhost", "127.0.0.1") {
+				return fmt.Errorf("subjectAltNames must not contain localhost, 127.0.0.1")
+			}
+		} else {
+			if stringSliceContains(c.SubjectAltNames, c.NodeIP) {
+				return fmt.Errorf("subjectAltNames must not contain node IP")
```

Comment:
```suggestion
				return fmt.Errorf("subjectAltNames must not contain node IP")
```
 
We should probably document why this should not be done (the SNI/client-go issues)

Reply:
Good point. Let me add the comment